### PR TITLE
Make netty and aeron dependencies optional

### PR DESCRIPTION
**Pull request by chbatey**
Refs #25169

I've tried this out with the cluster sample, error messages when missing:

```
info] Running (fork) sample.cluster.simple.SimpleClusterApp 
[error] Exception in thread "main" java.lang.IllegalStateException: Artery remoting is enabled but Aeron driver is not on the classpath, it must be added explicitly. See https://doc.akka.io/docs/akka/2.6/remoting-artery.html
[error] 	at akka.remote.RemoteActorRefProvider.checkClassOrThrow(RemoteActorRefProvider.scala:271)
```

```
info] Running (fork) sample.cluster.simple.SimpleClusterApp 
[error] Exception in thread "main" java.lang.IllegalStateException: Classic remoting is enabled but Netty is not on the classpath, it must be added explicitly. See https://doc.akka.io/docs/akka/2.6/remoting.html
[error] 	at akka.remote.RemoteActorRefProvider.checkClassOrThrow(RemoteActorRefProvider.scala:271)
[error] 	at akka.remote.RemoteActorRefProvider.checkNettyOnClassPath(RemoteActorRefProvider.scala:258)
[error] 	at akka.remote.RemoteActorRefProvider.init(RemoteActorRefProvider.scala:217)
```

Pom file for Akka remote:

```
        <dependency>
            <groupId>io.netty</groupId>
            <artifactId>netty</artifactId>
            <version>3.10.6.Final</version>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>io.aeron</groupId>
            <artifactId>aeron-driver</artifactId>
            <version>1.15.1</version>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>io.aeron</groupId>
            <artifactId>aeron-client</artifactId>
            <version>1.15.1</version>
            <optional>true</optional>
        </dependency>
```

I've ran multi jvm tests but need to check multi node tests in Jenkins

### CODE REVIEW

**Comment by chbatey** [comment]
I originally had the system shutdown via Coordinated shutdown with a new reason but all hell broke loose as other things still tried to start

**Comment by chbatey** [comment]
Assuming we're okay with checking in the docs links with 2.6 i'll create an issue to switch them to current after reviews

**Comment by hepin1989** [comment]
Will you split it to a new subproject?

**Comment by chbatey** [comment]
@hepin1989 no plans to, we'll remove the netty code in 2.7. They're both optional so you add the one you want.

File: akka-docs/src/main/paradox/remoting-artery.md
Comment (patriknw) [inline akka-docs/src/main/paradox/remoting-artery.md]: s/artery/Artery/

File: akka-docs/src/main/paradox/remoting.md
Comment (patriknw) [inline akka-docs/src/main/paradox/remoting.md]: do not *have to have*

File: akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: those are public api so that seems like low risk

File: project/Dependencies.scala
Comment (patriknw) [inline project/Dependencies.scala]: good, maybe add additional note up at the version numbers that it should be updated if the aeronVersion is updated so we use same as the aeron dependency

**Review by patriknw — APPROVED** [review]
LGTM, only something small...

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md
Comment (helena) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: depends on?

File: project/Dependencies.scala
Comment (helena) [inline project/Dependencies.scala]: Or Optional for no string usage

**Review by helena — APPROVED** [review]
Looks pretty good, one or two suggestions.

File: project/Dependencies.scala
Comment (chbatey) [reply project/Dependencies.scala]: :+1: will leave it to remain consistent with the rest of the file but we should switch them all

**Comment by chbatey** [comment]
PLS BUILD

File: project/Dependencies.scala
Comment (helena) [reply project/Dependencies.scala]: Once this merges I'll do https://github.com/akka/akka/issues/27040

**Comment by helena** [comment]
This has the same issue as one of my PRs
```scala
Permission denied, please try again.
Permission denied, please try again.
Permission denied (publickey,password).
```

**Comment by chbatey** [comment]
PLS BUILD

**Pull request merged**
